flink-connectors/flink-orc/src/test/java/org/apache/flink/orc/OrcTableSourceITCase.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.orc;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.io.DiscardingOutputFormat;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.java.BatchTableEnvironment;
+import org.apache.flink.test.util.MultipleProgramsTestBase;
+import org.apache.flink.types.Row;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+/**
+ * Tests both {@link OrcTableSource} and {@link RowOrcInputFormat}
```

Comment:
The tests for the `OrcInputFormat` should check all file related issues and most of the heavy ITCases:
- multiple splits
- multiple files of same schema
- tests for the different data types
- proper projection
- correct `getProducedType()` (without and with `setFieldMapping()` being called.
- proper exception in case of incorrect schema
- proper exception in case of incorrect projection

Some of these tests can be done in a lightweight unit test. Others need more heavy ITCases.

The `TableSource` tests should check that
- the projection is correctly pushed down (can be done with a test that checks the plan, see `TableTestBase`)
- the OrcInputFormat is correctly created and configured. 
- we do not need to test if the Table API correctly supports nested types. These checks should be done by the Table API itself.
- an end-to-end ITCase with projection push-down